### PR TITLE
feat: make broker_heartbeat configurable via config.yaml

### LINKED ISSUE #25: broker_heartbeat
This issue I make after Lukas Frey asked the the following question:

---

I'm running the bigpicture-slidetap-xlsx stack on a Linux machine and experiencing repeated crashes of the taskservice during large batch processing (~1300 WSI files).

The taskservice consistently shuts down gracefully (no hard crash) after processing for some time. The shutdown sequence always looks like this:

  [DEBUG/MainProcess] result handler: all workers terminated
  [DEBUG/MainProcess] | Worker: Stopping Hub...
  [DEBUG/MainProcess] | Consumer: Shutdown Heart...
  ...
  taskservice exited with code 1

The tasks themselves succeed (post_process_image), but some take very long (up to 4500 seconds per image due to SVS → DICOM conversion). After the crash, the batch is left in IMAGE_POST_PROCESSING status and has to be manually reset to IMAGE_POST_PROCESSING_COMPLETE in the database before processing can be resumed via the UI.

I suspect the issue is related to the RabbitMQ heartbeat timeout: when a ForkPoolWorker is busy with a long-running task, the main process seems unable to send heartbeats, causing the connection to drop and the worker to shut down. The heartbeat interval negotiated between Celery and RabbitMQ is 120 seconds, which is shorter than some individual tasks.

I have tried the following without success:
- Setting RABBITMQ_SERVER_ADDITIONAL_ERL_ARGS=-rabbit heartbeat 0
- Adding --without-heartbeat and --heartbeat-interval 0 to the Celery worker command
- Setting CELERY_BROKER_HEARTBEAT=0 as an environment variable

None of these fully disabled the heartbeat (it stays at 120 seconds).

### PR DESCRIPTION
**Pull request by JeroenUMC**
Previously broker_heartbeat was hardcoded to 120 in CeleryConfig.dict_config. This change adds a broker_heartbeat field to CeleryConfig (default 120) that can be overridden in the celery section of config.yaml. Setting it to 0 disables the heartbeat entirely, which is useful for environments where the AMQP heartbeat causes connection drops.

Closes #25

### CODE REVIEW

File: slidetap-app/tests/test_celery_config.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+def make_parser(config: dict, broker_url: str = "amqp://localhost") -> ConfigParser:
+    """Helper to create a ConfigParser with an injected broker URL."""
+    parser = ConfigParser(config=config)
+    # Patch get_env so tests don't require environment variables
+    parser.get_env = lambda key, default=None: broker_url if key == "SLIDETAP_BROKER_URL" else default  # type: ignore[method-assign]
+    return parser
+
+
+@pytest.mark.unittest
+class TestCeleryConfigBrokerHeartbeat:
```
Comment (erikogabrielsson) [inline slidetap-app/tests/test_celery_config.py:28]: Should be named TestCeleryConfig

File: slidetap-app/tests/test_celery_config.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    def test_default_heartbeat_when_no_celery_section(self):
```
Comment (erikogabrielsson) [inline slidetap-app/tests/test_celery_config.py:29]: Follow the explicit Arrange-Act-Assert pattern used in other unit tests

File: slidetap-app/tests/test_celery_config.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import pytest
+from slidetap.config import CeleryConfig, ConfigParser
+
+
+def make_parser(config: dict, broker_url: str = "amqp://localhost") -> ConfigParser:
```
Comment (erikogabrielsson) [inline slidetap-app/tests/test_celery_config.py:19]: Use pytest fixture

**Review by erikogabrielsson — COMMENTED** [review]
A few suggestions before merging. The core change is solid; these are polish items.

### 1. Pick `Optional[int]` or `int`, not both

```python
broker_heartbeat: Optional[int] = 120
```

The `Optional` annotation suggests `None` is meaningful (in Celery, `None` defers to the library default; `0` explicitly disables). But the parse path can never produce `None` — `get_yaml_or_default("broker_heartbeat", 120)` always returns an `int`. Pick one:

- **`Optional[int] = None`** if `None` should mean "let Celery decide" — then change the parse default to `None` so YAML omission preserves that behavior.
- **`int = 120`** if `None` isn't meaningful — `0` already covers "disable".

I'd lean toward `Optional[int] = None` since it gives users three distinct states (None / 0 / positive int) and is more honest about Celery's behavior.

### 2. Add a field-level docstring

The PR description explains what `0` does, but the code doesn't. A `#:` comment or inline docstring helps future readers:

```python
#: Celery AMQP broker heartbeat in seconds. ``0`` disables it
#: (useful when intermediate proxies drop idle TCP connections).
broker_heartbeat: Optional[int] = None
```

### 3. Add a `None` test

If you go with `Optional[int]`, add a test that exercises `None` (either via explicit YAML `null` or YAML omission with `None` default), and assert it propagates to `dict_config`. Otherwise the type annotation lies about the API.

### 4. Simplify the test fixture

`main` now has a small prep change (`e30acd0` *Make config parser testable*) that adds an injectable `env` param to `ConfigParser.__init__`. After rebasing, you can drop the `parser.get_env = lambda ...` monkey-patch and `# type: ignore[method-assign]` lines, and simplify the fixture to:

```python
def _parser(celery: Dict[str, Any] | None = None) -> ConfigParser:
    return ConfigParser(
        config={"celery": celery} if celery is not None else {},
        env={"SLIDETAP_BROKER_URL": "amqp://localhost"},
    )
```

The two existing fixtures (`empty_config`, `config_with_celery`) collapse to one helper and the per-test `decoy.when(...).then_enter_with(...)`-style stubbing is no longer needed. The factory-as-fixture pattern (`return _make  # type: ignore[return-value]`) also goes away.

**Comment by erikogabrielsson** [comment]
Closing this, as the project no longer uses cellery.
